lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,531 @@
+            'OBJECT': str,
+            'REAL': float,
+            'TIMESTAMP': str,
+            'VARCHAR': str,
+            'XML': str,
+            'SAMPLE': str
+        }
+        logging.debug('TYPE CHECKING OF THE SAMPLE')
+        logging.debug('Setting up comparison dict')
+        types_df = o.get_sample_properties(self.type)
+        types_dict = dict(zip(types_df.code, types_df.dataType))
+        logging.debug(types_dict)
+        types_dict: dict = {k.lower(): conv_dict[v] for k, v in types_dict.items()}
+
+        # Check if all keys are a subset of all possible keys
+        logging.debug('Checking if keys are subset of defined parameters')
+        if set(self.metadata.keys()).issubset(set(types_dict.keys())):
+            for key, val in self.metadata.items():
+                # Check if all values have the correct data type
+                if not isinstance(val, types_dict[key]):
+                    raise ValueError(
+                        f'Type Checker: Type Mismatch.'
+                        f'Entry "{key}" is of Type {type(val)} and should be {types_dict[key]}')
+                    # Check if float values are not NaN
+                if isinstance(val, float) and isnan(val):
+                    raise ValueError(
+                        f'Type Checker: NaN values are not accepted by Openbis.'
+                        f'Entry "{key} has a value "{val}".')
+        else:
+            # Keys in metadata and not in object definition -> raises an error for every such key
+            for key in self.metadata.keys():
+                if key.lower() not in types_dict.keys():
+                    raise ValueError(
+                        f'Type Checker: Unexpected parameter.'
+                        f'Key "{key}" is not in {self.type} defined metadata params')
+
+        # Checking if all values are defined
+        needed_attrs = ['name', 'type', 'collection', 'space', 'project']
+        for attr, value in self.__dict__.items():
+            if attr in needed_attrs:
+                if not value:
+                    raise ValueError(
+                        f'Type Checker: Undeclared Attribute.'
+                        f'Attribute "{attr}" has not been delcared')
+
+        # Printing warnings if parents are empty, does not break the function
+        if not self.parents:
+            logging.warning(
+                'Type Checker: Parents attribute undeclared')
+
+        # If you got to this line no game breaking bugs were found
+        logging.info('Type Checker: Types are correctly set')
+
+    @staticmethod
+    def load_sample(o: Interbis, sample_identifier: str):
+        """Loads an ExpStep from an experimental step in the datastore with its properties
+
+        Args:
+            o (Interbis): currently running openBIS interface
+            sample_identifier (str): identifier of the sample
+
+        Returns:
+            ExpStep: ExpStep object containing metadata of the sample
```

Comment:
Is that an ExpStep object (with metada=[author=string, date=date, diameter=float]) or a sample object with metadata = (type=youngsmodulus, author="Max Mustermann", diameter=25). 
If I understand the code correctly, "Loads an ExpStep (the metadata structure with the corresponding data types) from a dataset in the openBIS instance with the sample_identifier str"

Reply:
No the load_sample method will load an "instantiated" ExpStep/Sample so something you had already defined and uploaded to the datastore. The method you are describing would be (more or less) done by the Interbis get_sample_type_properties method.